Delete unneeded files

Psirus opened the pull request:
I've deleted a couple of folders/files that seem unnecessary and no longer in use.

Specifically, I think basically everyone is installing vim plugins using a plugin manager these days, so all the debian and vimball stuff are gone. We should probably change the documentation to reflect that.

Additionally, there was documentation copied from orgmode itself, which a) I don't see the need to duplicate here and b) is probably outdated by now.

Let me know what you think.

XVilka commented:
Can you please fix the documentation here then too? https://github.com/jceb/vim-orgmode/blob/master/doc/orgguide.txt#L1457

Psirus commented:
I've removed the old documentation, and added examples for the Vim 8 and pathogen package managers, so this should now also close #294, #301 and #135.
